src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/world/ModuleSurround.kt

```diff
@@ -0,0 +1,403 @@
+     */
+    private val disableOnXZSpeed by boolean("disableOnXZSpeed", false)
+
+    /**
+     * Replaces broken block instantly.
+     *
+     * Note: requires the rotation mode "None" in the block placer
+     */
+    private val instant by boolean("Instant", true)
+
+    /**
+     * Protects the surround against being blocked by crystals on destruction.
+     *
+     * Requires the crystal destroyer in the placer to be active.
+     */
+    private object Protect : ToggleableConfigurable(this, "Protect", true) {
+
+        /**
+         * At what destroy stage, actions should be taken.
+         */
+        private val minDestroyProgress by int("MinDestroyProgress", 4, 0..9, "stage")
+
+        /**
+         * Builds an extra layer around the surround blocks (slice):
+         *     p
+         *   x p x
+         *     x
+         * will become:
+         *   x p x
+         * x x p x x
+         *     x
+         *
+         * X = obsidian
+         * p = the players hitbox
+         */
+        @Suppress("SpellCheckingInspection", "GrazieInspection")
+        object ExtraLayer : ToggleableConfigurable(this, "ExtraLayer", true) {
+
+            /**
+             * Will place even more block (top view):
+             *   x
+             * x p x
+             *   x
+             * will become:
+             * x x x
+             * x p x
+             * x x x
+             *
+             * X = obsidian
+             * p = the players hitbox
+             */
+            val corners by boolean("Corners", false)
+
+        }
+
+        init {
+            tree(ExtraLayer)
+        }
+
+        val broken = mutableSetOf<BlockPos>()
+
+        /**
+         * With a higher priority so that it runs before [CrystalDestroyFeature].
+         */
+        @Suppress("unused")
+        private val tickHandler = handler<GameTickEvent>(priority = 10) {
+            if (!placer.crystalDestroyer.enabled || addExtraLayerBlocks) {
+                return@handler
+            }
+
+            broken.clear()
+            placer.blocks.filter { !it.value }.keys.forEach { pos ->
+                val breakingProgressions = mc.worldRenderer.blockBreakingProgressions[pos.asLong()] ?: return@forEach
+
+                val breakingInfo = breakingProgressions.lastOrNull { it.actorId != player.id } ?: return@forEach
+                val stage = breakingInfo.stage
+                if (stage < minDestroyProgress) {
+                    return@forEach
+                }
+
+                if (ExtraLayer.enabled && stage > 0) {
+                    broken.add(pos)
+                }
+
+                val blockedResult = pos.isBlockedByEntitiesReturnCrystal()
+                val crystal = blockedResult.value() ?: return@forEach
+
+                placer.crystalDestroyer.currentTarget = crystal
+                if (placer.crystalDestroyer.currentTarget == crystal) {
+                    return@handler
+                }
+            }
```

Comment:
I don't quite understand what this code does. It could be a bit more imperative.